### ISSUE TITLE
Ability to render `html` (or some rich content) in balloon.

### ISSUE THREAD

**Issue by jordwalke** [macvim-dev/macvim#218]
I wanted to be able to control the font that is shown for a particular tooltip balloon in MacVim. The use case is for showing compiler errors in monospaced font (otherwise, they render terribly). Then, I quickly realized that I only want *some* of the tooltip to be monospace and some to be rendered with standard text font. I also wanted greater control over the content in general, and realized that the best way to do this would be to allow the tooltip content to accept arbitrary HTML and have that be rendered using WebKit. I believe this could be done entirely in MacVim and not concern Vim itself. When writing a plugin that shows tooltips, you would want to check if you're running inside of MacVim to determine if you should show rich HTML content vs plain text.

**Comment by splhack** [comment]
Why don't you use this? https://github.com/rhysd/neovim-component

**Comment by jordwalke** [comment]
It would be nice to write plugins that can work with MacVim since many people will not use `neovim-component`. I'm approaching this as a Vim plugin author for the moment.

**Comment by splhack** [comment]
In terms of install user base, just use https://atom.io/ is one of the best solution.
I'd like to keep MacVim would be simple, fast as Vim. I don't want the feature if it will affect the performance.

**Comment by jordwalke** [comment]
@splhack I agree, I wouldn't want to compromise performance either. The feature request assumes there wouldn't be any downside to supporting rich balloon content, aside from the fact that it has to be implemented.

**Comment by eirnym** [comment]
It's possible to implement this feature as a plugin, not a core MacVim functionality.

**Comment by ychin** [comment]
@jordwalke I'm currently considering this feature as part of planning out MacVim featureset for future releases. Do you still have an interest in this? Trying to gauge potential uses of a feature like this and what you may want to use it for.

For example, we could pop up a balloon with rich content or an image / html page, or we could show custom UI as a separate panel, and there have been people requesting a file view before (#1176), so I'm trying to think through potential use cases, implementation costs/complexities, and whatnot. Feel free to just chime in and maybe talk through your thoughts.
